opta/commands/upgrade.py

```diff
@@ -0,0 +1,52 @@
+import os
+
+import click
+import requests
+from colored import attr, fg
+
+from opta.constants import OPTA_INSTALL_URL, successfull_upgrade
+from opta.nice_subprocess import nice_run
+from opta.upgrade import check_version_upgrade
+from opta.utils import logger
+
+TEMP_INSTALLATION_FILENAME = "opta_installation.sh"
+
+
+def _make_installation_file() -> None:
+    logger.debug(f"Querying {OPTA_INSTALL_URL}")
+    resp = requests.get(OPTA_INSTALL_URL)
+    resp.raise_for_status()
+    with open(TEMP_INSTALLATION_FILENAME, "w") as file:
+        file.write(resp.text)
+    nice_run(["chmod", "777", TEMP_INSTALLATION_FILENAME])
+
+
+def _upgrade_successful() -> None:
+    open(successfull_upgrade, "w").close()
+
+
+def _cleanup_installation_file() -> None:
+    if os.path.isfile(TEMP_INSTALLATION_FILENAME):
+        os.remove(TEMP_INSTALLATION_FILENAME)
+
+
+@click.command()
+def upgrade() -> None:
```

Comment:
do you think we can add some unit test to test upgrade()?
this new part is not covered by existing tests right?

Reply:
added